realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java

```diff
@@ -39,97 +36,84 @@
 
-        testRealm.beginTransaction();
-        Owner owner = testRealm.createObject(Owner.class);
+        realm.beginTransaction();
+        Owner owner = realm.createObject(Owner.class);
         owner.setName("Owner");
-        for (int i = 0; i < TEST_OBJECTS; i++) {
-            Dog dog = testRealm.createObject(Dog.class);
+        for (int i = 0; i < TEST_SIZE; i++) {
+            Dog dog = realm.createObject(Dog.class);
             dog.setName("Dog " + i);
             owner.getDogs().add(dog);
         }
-        testRealm.commitTransaction();
+        realm.commitTransaction();
+        collection = owner.getDogs();
     }
 
     @After
     public void tearDown() throws Exception {
-        if (testRealm != null) {
-            testRealm.close();
+        if (realm != null) {
+            realm.close();
         }
     }
 
     private RealmList<Dog> createNonManagedDogList() {
         RealmList<Dog> list = new RealmList<Dog>();
-        for (int i = 0; i < TEST_OBJECTS; i++) {
+        for (int i = 0; i < TEST_SIZE; i++) {
             list.add(new Dog("Dog " + i));
         }
         return list;
     }
 
     private RealmList<Dog> createDeletedRealmList() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
+        Owner owner = realm.where(Owner.class).findFirst();
         RealmList<Dog> dogs = owner.getDogs();
 
-        testRealm.beginTransaction();
-        owner.removeFromRealm();
-        testRealm.commitTransaction();
+        realm.beginTransaction();
+        owner.deleteFromRealm();
+        realm.commitTransaction();
         return dogs;
     }
 
-    // Check that all methods work correctly on a empty RealmList
-    private void checkMethodsOnEmptyList(Realm realm, RealmList<Dog> list) {
-        realm.beginTransaction();
-        for (int i = 0; i < 4; i++) {
             //noinspection TryWithIdenticalCatches
-            try {
-                switch (i) {
-                    case 0: list.get(0); break;
-                    case 1: list.remove(0); break;
-                    case 2: list.set(0, new Dog()); break;
-                    case 3: list.move(0, 0); break;
-                    default: break;
-                }
-                fail();
-            } catch (IndexOutOfBoundsException ignored) {
-            } catch (RealmException ignored) {
-            }
-        }
-        realm.cancelTransaction();
-
-        assertEquals(0, list.size());
-        assertNull(list.first());
-        assertNull(list.last());
-    }
-
     /*********************************************************
-     * Non-Managed mode tests                                *
+     * Un-managed mode tests                                *
      *********************************************************/
 
+    @Test(expected = IllegalArgumentException.class)
+    public void constructor_nonManaged_null() {
+        AllTypes[] args = null;
+        new RealmList<AllTypes>(args);
```

Comment:
Actually doing that will cause javac to complain. That was what lead to this. I'll suppress the warning in IntelliJ instead.